oauth2client/client.py

```diff
@@ -841,9 +785,8 @@ def _refresh(self, http_request):
         if not self.store:
             self._do_refresh_request(http_request)
         else:
-            self.store.acquire_lock()
-            try:
-                new_cred = self.store.locked_get()
+            with self.store.lock():
```

Comment:
Massive :+1: to this pattern.

I'd wager that you need to very clearly set expectations for whether `get/set/delete` are expected to happen while locked or while not. At the very least someone should rewrite the docstrings for existing implementations to make this clear.